src/sqlfluff/core/rules/base.py

```diff
@@ -432,6 +504,17 @@ class BaseRule:
     _adjust_anchors = False
     targets_templated = False
 
+    # Lint loop / crawl behavior. When appropriate, rules can (and should)
+    # override these values to make linting faster.
+    recurse_into = True
+    needs_raw_stack = True
+    # Rules can override this to specify "post". "Post" rules are those that are
+    # not expected to trigger any downstream rules, e.g. capitalization fixes.
+    # They run on two occasions:
+    # - On the first loop of the main phase
+    # - In a second linter loop after the main rules run
+    lint_phase = "main"
+
```

Comment:
Anything to be said for a memory_factory prop?
then we can instantiate the memory at the right times in the crawl. and save that first bit of boilerplate? And encourage typing?

Reply:
I'd definitely like to streamline use of memory some time. Probably a separate issue/PR (performance vs tech debt/design improvement).